content/blog/2020-09-17-delivering-learnr-tutorials-in-a-package/index.Rmarkdown

```diff
@@ -0,0 +1,322 @@
+
+    ```{r, eval = TRUE, echo = FALSE}
+    knitr::include_graphics("pkg-build.png")
+    ```
+
+    * Your project directory should end up looking something like this:
+
+    ```{r, eval = TRUE, echo = FALSE}
+    knitr::include_graphics("pkg-directory.png")
+    ```
+
+
+
+### Add files and directories for a tutorial
+
+Now that the package structure is in place, it's time to add tutorial content. Again, the usethis package will do the dirty work for us to create the folders and files we need for a learnr tutorial within our package.
+
+1. **Create the learnr tutorial** and open it for editing when you run `usethis::use_tutorial("<tutorial-name>", "<Tutorial Title>")`:
+
+    ```{r, eval = FALSE}
+    usethis::use_tutorial("lesson1", "Your First Lesson", open = interactive())
+    ```
+
+    * The first argument is the basename for the learnr R Markdown file you are creating. You do not need the `.Rmd` file extension here.
+
+    *  The second argument is what you want the user-facing title of the tutorial to be.
+
+    * This creates an `inst/` directory with some subfolders. If you're not new to creating learnr tutorials, the folder structure here will look familiar to you -- there's nothing different about learnr tutorial directories inside and outside of packages.
+
+      You can open up `inst/` and its subsequent subfolders until you reach the `.Rmd` file you created:
+
+      ```
+      /inst/tutorials/<tutorial-name>/<tutorial-name.Rmd>
+      ```
+
+    ```{r, eval = TRUE, echo = FALSE}
+    knitr::include_graphics("pkg-inst.png")
+    ```
+
+1. **Edit the open `.Rmd` file** and customize your tutorial as you normally would.
+
+1. **Repeat** for each additional tutorials you want to include in this package.
+
+
+
+### Build, install, and use your tutorial package
+
+Time to build the package and install it in your R system library.
+
+1. __Navigate to the *Build* pane__ of your RStudio IDE. Then __click *Install and Restart*__. Alternatively, run:
+
+    ```{r, eval = FALSE}
+    devtools::install()
+    ```
+
+    Believe it or not, you've now made a package with a learnr tutorial inside!
+
+    You should test that your package has been properly created by calling the function that will load your tutorial from your installed package:
+
+1.  **Run your tutorial** with `learnr::run_tutorial("<tutorial-name>", "<package-name>")`.
+
+    ```{r, eval = FALSE}
+    learnr::run_tutorial("lesson1", "tutorialpackage")
+    ```
+
+    * The tutorial name here refers to the name of the `.Rmd` basename, not the user-facing title.
+    * The live tutorial will open up in a local browser window (*woot!*)
+    * This code is also how your users will be able to run your tutorial once they have installed your package.
```

Comment:
And following on, I'd definitely recommend changing this -- I think users running from the tutorials pane is the thing to promote here